pkg/cloudprovider/plugins/azure/azure_resourcegraph_vmTable.go

```diff
@@ -152,19 +152,24 @@ func getVirtualMachineTable(resourceGraphAPIClient azureResourceGraphWrapper, qu
 	if err != nil {
 		return nil, 0, fmt.Errorf("error invoking Azure resource graph query: %v", err)
 	}
-	if data == nil {
+	if len(data) == 0 {
 		return []*virtualMachineTable{}, 0, nil
 	}
 
 	var virtualMachines []*virtualMachineTable
-	virtualMachineRows := data.([]interface{})
-	for _, virtualMachineRow := range virtualMachineRows {
-		var virtualMachine virtualMachineTable
-		err = customDecode(virtualMachineRow.(map[string]interface{}), &virtualMachine)
-		if err != nil {
-			return nil, 0, fmt.Errorf("error decoding Azure resource graph query's response: %v", err)
+	for _, page := range data {
+		if page == nil {
```

Comment:
What is the scenario when data is there, however page is nil? its not clear to me..

Reply:
As discussed, Updated the PR to have data field include the actual data instead of presenting in the format for individual page.